### PR TITLE
ocamlPackages.morbig: init at 0.11.0

### PR DESCRIPTION
**Pull request by Niols**
###### Description of changes

[Morbig](https://github.com/colis-anr/morbig/) is a trustworthy static parser for shell scripts written in the POSIX shell script language. It parses the scripts statically, that is without executing them, and constructs a concrete syntax tree for each of them. The concrete syntax trees are built using constructors according to the shell grammar of the POSIX standard.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 226101` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by ulrikstrid — APPROVED** [review]
Looks good, the executable seems to work as well. Thank you!

File: pkgs/development/ocaml-modules/morbig/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  nativeBuildInputs = with ocamlPackages; [
+    menhir
+  ];
+
+  propagatedBuildInputs = with ocamlPackages; [
+    menhirLib
+    ppx_deriving_yojson
+    visitors
+    yojson
+  ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/morbig/default.nix:25]: These packages should be taken individually as arguments instead of the `ocamlPackagse` set.

Comment (Niols) [reply pkgs/development/ocaml-modules/morbig/default.nix:25]: I can open a PR to fix that. Do you have a pointer to something that explains the difference?

Comment (vbgl) [reply pkgs/development/ocaml-modules/morbig/default.nix:25]: Currently, `ocaml-ng.ocamlPackages_4_12.morbig` fails to build.

Comment (Niols) [reply pkgs/development/ocaml-modules/morbig/default.nix:25]: cf https://github.com/NixOS/nixpkgs/pull/226131

**Pull request merged**